specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+first frame in the queue, or is preceded by a closing frame of a previous channel.
+
+Note that these rules are in contrast to pre-Holocene rules, where out of order frames were
+buffered. Pre-Holocene, frame validity checks were only done at the Channel Bank stage. Performing
+these checks already at the Frame Queue stage leads to faster discarding of invalid frames, keeping
+the memory consumption of any implementation leaner.
+
+## Channel Bank
+
+Because channel frames have to arrive in order, the Channel Bank becomes much simpler and only
+holds at most a single channel at a time.
+
+### Pruning
+
+Pruning is vastly simplified as there's at most only one open channel in the channel bank. So the
+channel bank's queue becomes effectively a staging slot for a single channel, the _staging channel_.
+The `MAX_CHANNEL_BANK_SIZE` parameter becomes ineffective and the size of a channel is bounded, as
+before, during decompression to be at most of size `MAX_RLP_BYTES_PER_CHANNEL`.
+
+### Timeout
+
+The timeout is applied as before, just only to the single staging channel.
+
+### Reading & Frame Loading
+
+The frame queue is guaranteed to hold ordered and contiguous frames, per channel. So reading and
+frame loading becomes simpler in the channel bank:
+
+- A first frame for a new channel starts a new channel as the staging channel.
+  - If there already is an open, non-completed staging channel, it is dropped and replaced by this
+  new channel. This is consistent with how the frame queue drops all frames of an non-closed channel
+  upon the arrival of a first frame for a new channel.
+- If the current channel is timed-out, but not yet pruned, and the incoming frame would be the next
+correct frame for this channel, the frame and channel are dropped, including all future frames for
+the channel that might still be in the frame queue. Note that the equivalent rule was already
+present pre-Holocene.
+
+## Span Batches
+
+Partial Span Batch Validity changes the atomic validity model of [Span Batches](../delta/span-batches.md).
+In Holocene, a span batch is treated as an optional stage in the derivation pipeline that sits
+before the batch queue, so that the batch queue pulls singular batches from this previous Span Batch
+stage. When encountering an invalid singular batch, it is dropped, as is the remaining span batch
+for consistency reasons. We call this _forwards-invalidation_. However, we don't
+_backwards-invalidate_ previous valid batches that came from the same span batch, as pre-Holocene.
+
+An implementation may choose to put the current span batch, from which singular batches are derived
+from, into a separate stage, or process it within the batch queue under the new rules.
+
+When a batch derived from the current staging channel is a singular batch, it is directly forwarded
+to the batch queue. Otherwise, it is set as the current span batch in the span batch stage. The
```

Comment:
I could be missing something, but this reads to me like you are offering two implementation hints but then proceeding to describe just one of those implementations in more detail? If so, I think it could be made clearer.

Reply:
Picked up elsewhere, but also wondering if implementation related material belongs here? Seems very useful though.

Reply:
On second thought, I think this is such a low-level implementation detail that I will just remove this sentence and not even move it below to the "Implementation Considerations" section.